src/endpoints/CMakeLists.txt

```diff
@@ -1,11 +1,37 @@
 option(IRODS_ENABLE_GENQUERY2 "Enables GenQuery2 support. GenQuery2 package must be installed." NO)
 
+set(IRODS_HTTP_SHARED_API_OPERATIONS shared_api_operations)
+
+add_library(
+  ${IRODS_HTTP_SHARED_API_OPERATIONS}
+  OBJECT
+  shared_api_operations.cpp)
+
+target_compile_definitions(
+  ${IRODS_HTTP_SHARED_API_OPERATIONS}
+  PRIVATE
+  ${IRODS_COMPILE_DEFINITIONS}
+  ${IRODS_COMPILE_DEFINITIONS_PRIVATE})
+
+target_include_directories(
+  ${IRODS_HTTP_SHARED_API_OPERATIONS}
+  PRIVATE
+  ${IRODS_INCLUDE_DIRS}
+  ${IRODS_EXTERNALS_FULLPATH_BOOST}/include
+  ${IRODS_EXTERNALS_FULLPATH_FMT}/include
+  ${IRODS_EXTERNALS_FULLPATH_SPDLOG}/include
+  CURL::libcurl
+  #nlohmann_json::nlohmann_json
```

Comment:
is this coming soon?   or should be removed?

Reply:
I think that is what we want, but it doesn't work for some reason. I'm going to look into it to try and learn why before I reach out to @SwooshyCueb.

Reply:
We don't want to put targets in `target_include_directories` at all. They go in `target_link_libraries`.

Reply:
The commented out code certainly didn't work. Things only compiled when using `${IRODS_EXTERNALS_FULLPATH_JSON}/include`.

I'm guessing we aren't doing what CURL is doing then? I feel like I've seen `nlohmann_json::nlohmann_json` in an include directive before.

Reply:
`CURL::libcurl` should be in `target_link_libraries`, too. Passing a target to `target_link_libraries` takes care of includes as well as any library linkages. You shouldn't need to *ever* reference `IRODS_EXTERNALS_FULLPATH_JSON` unless something has gone horribly wrong.

Reply:
Oh. I had no idea `target_link_libraries` handled include paths. I'll remove that and see what happens.

Reply:
Done.